pkg/network/protocols/http/ebpf_main.go

```diff
@@ -94,6 +94,13 @@ var tailCalls = []manager.TailCallRoute{
 			EBPFFuncName: "socket__http_filter",
 		},
 	},
+	{
+		ProgArrayName: protocolDispatcherProgramsMap,
+		Key:           uint32(ProtocolHTTP2),
+		ProbeIdentificationPair: manager.ProbeIdentificationPair{
+			EBPFFuncName: "socket__http2_filter",
+		},
+	},
```

Comment:
add the tail call only if http2 is enabled
another option is to exclude it is http2 is disabled